src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -309,7 +309,14 @@ def assemble_json(ids):
             # place the ID parts into the correct property lists
             parts = parse_resource_id(val)
             for arg in combined_args:
-                getattr(namespace, arg.name).append(parts[arg.type.settings['id_part']])
+                id_part = arg.type.settings.get('id_part')
+                id_value = parts.get(id_part, None)
+                if id_value is None:
+                    alias = arg.type.settings.get('options_list', None)
+                    argument_name = alias[0] if alias is not None else arg.name
```

Comment:
`arg.name` is like `vm_name` which is the name of arg in SDK. Consider transforming the arg to CLI form, like  https://github.com/microsoft/knack/blob/44ee1799c891213c90916f186ab9aee7b9ff8344/knack/arguments.py#L66

```py
'--{}'.format(self.options['dest'].replace('_', '-'))
```